tensorboard/backend/event_processing/plugin_event_accumulator.py

```diff
@@ -510,11 +511,16 @@ def _MaybePurgeOrphanedData(self, event):
     def _CheckForRestartAndMaybePurge(self, event):
         """Check and discard expired events using SessionLog.START.
 
-        Check for a SessionLog.START event and purge all previously seen events
-        with larger steps, because they are out of date. Because of supervisor
-        threading, it is possible that this logic will cause the first few event
-        messages to be discarded since supervisor threading does not guarantee
-        that the START message is deterministically written first.
+        The first SessionLog.START event in a run indicates the start of a
+        supervisor session. Subsequent SessionLog.START events indicate a
+        *restart*, which may need to preempt old events. This method checks
+        for a session restart event and purges all previously seen events whose
+        step is larger than or equal to this event's step.
+
+        Because of supervisor threading, it is possible that this logic will
+        cause the first few event messages to be discarded since supervisor
+        threading does not guarantee that the START message is deterministically
+        written first.
```

Comment:
I might be mistaken, but isn't this 2nd paragraph no longer relevant?    The "first few event messages" can still be discarded, but that is because of multiple START messages appearing, not because of a START message appearing after the first few messages.

Reply:
Hmm, I’m not sure that I follow. I interpreted this as meaning, “at the
start of a supervisor invocation, the order of the `START` event and the
initial summaries may not be deterministic”. So if we have

```
file_version
graph_def [step: 0]
START [step: 0]
scalar [step: 0]
scalar [step: 1]
// preempted!
scalar [step: 2]
START [step: 2]
scalar [step: 3]
```

where the step-2 `START` should have been written before the step-2
scalar, but wasn’t because of “threading reasons”, then I think that
this paragraph is still relevant.

That said, I did some light digging and wasn’t able to find more details
about the circumstances that this refers to, so I could well be missing
something.